### PR TITLE
Do not expose FS path on front.

### PR DESCRIPTION
**Pull request by trasher**
Fix some typos.

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #12785

### CODE REVIEW

**Comment by trasher** [comment]
This PR introduces new strings to translate, I'm not sure it should be part of 10.0.6.

**Review by orthagh — APPROVED** [review]

File: src/Document.php [hunk 1]
```diff
@@ -182,15 +182,21 @@ public function cleanDBonPurge()
...
+                        $this->fields["filepath"]
                     ));
                 } else {
-                    Session::addMessageAfterRedirect(
+                    Toolbox::logError(
                         sprintf(
                             __('Failed to delete the file %s'),
                             GLPI_DOC_DIR . "/" . $this->fields["filepath"]
                         ),
+                    );
```
Comment (cedric-anne) [inline src/Document.php:194]:
```suggestion
                    trigger_error(
                        sprintf(
                            __('Failed to delete the file %s'),
                            GLPI_DOC_DIR . "/" . $this->fields["filepath"]
                        ),
                        E_USER_WARNING
                    );
```

File: src/Document.php [hunk 2]
```diff
@@ -1157,8 +1163,11 @@ public function moveUploadedDocument(array &$input, $filename)
         }
 
         if (!is_file($fullpath)) {
-            Session::addMessageAfterRedirect(
+            Toolbox::logError(
                 sprintf(__('File %s not found.'), $fullpath),
+            );
```
Comment (cedric-anne) [inline src/Document.php:1168]:
```suggestion
            trigger_error(
                sprintf(__('File %s not found.'), $fullpath),
                E_USER_WARNING
            );
```

File: src/Document.php [hunk 3]
```diff
@@ -1186,16 +1195,22 @@ public function moveUploadedDocument(array &$input, $filename)
...
             } else {
                // TRANS: %1$s is the curent filename, %2$s is its directory
-                Session::addMessageAfterRedirect(
+                Toolbox::logError(
                     sprintf(
                         __('Failed to delete the file %1$s (%2$s)'),
                         $input['current_filename'],
                         GLPI_DOC_DIR . "/" . $input['current_filepath']
+                    )
+                );
```
Comment (cedric-anne) [inline src/Document.php:1209]:
```suggestion
                trigger_error(
                    sprintf(
                        __('Failed to delete the file %1$s (%2$s)'),
                        $input['current_filename'],
                        GLPI_DOC_DIR . "/" . $input['current_filepath']
                    ),
                    E_USER_WARNING
                );
```

File: src/Document.php [hunk 4]
```diff
@@ -1257,8 +1272,11 @@ public static function moveDocument(array &$input, $filename)
         }
 
         if (!is_file($fullpath)) {
+            Toolbox::logError(
+                sprintf(__('File %s not found.'), $fullpath)
+            );
```
Comment (cedric-anne) [inline src/Document.php:1277]:
```suggestion
            trigger_error(
                sprintf(__('File %s not found.'), $fullpath),
                E_USER_WARNING
            );
```

File: src/Document.php [hunk 5]
```diff
@@ -1287,16 +1305,22 @@ public static function moveDocument(array &$input, $filename)
...
             } else {
                // TRANS: %1$s is the curent filename, %2$s is its directory
-                Session::addMessageAfterRedirect(
+                Toolbox::logError(
                     sprintf(
                         __('Failed to delete the file %1$s (%2$s)'),
                         $input['current_filename'],
                         GLPI_DOC_DIR . "/" . $input['current_filepath']
+                    )
+                );
```
Comment (cedric-anne) [inline src/Document.php:1319]:
```suggestion
                trigger_error(
                    sprintf(
                        __('Failed to delete the file %1$s (%2$s)'),
                        $input['current_filename'],
                        GLPI_DOC_DIR . "/" . $input['current_filepath']
                    ),
                    E_USER_WARNING
                );
```

File: src/Document.php [hunk 6]
```diff
@@ -1388,16 +1412,22 @@ public static function uploadDocument(array &$input, $FILEDESC)
...
```
Comment (cedric-anne) [inline src/Document.php:1426]:
```suggestion
                trigger_error(
                    sprintf(
                        __('Failed to delete the file %1$s (%2$s)'),
                        $input['current_filename'],
                        GLPI_DOC_DIR . "/" . $input['current_filepath']
                    ),
                    E_USER_WARNING
                );
```

File: src/Document.php [hunk 7]
```diff
@@ -1453,10 +1483,15 @@ public static function getUploadFileValidLocationName($dir, $sha1sum)
         }
 
         if (!is_dir(GLPI_DOC_DIR)) {
-            Session::addMessageAfterRedirect(
+            Toolbox::logError(
                 sprintf(
                     __("The directory %s doesn't exist."),
                     GLPI_DOC_DIR
+                )
+            );
```
Comment (cedric-anne) [inline src/Document.php:1491]:
```suggestion
            trigger_error(
                sprintf(
                    __("The directory %s doesn't exist."),
                    GLPI_DOC_DIR
                ),
                E_USER_WARNING
            );
```

File: src/Document.php [hunk 8]
```diff
@@ -1471,15 +1506,21 @@ public static function getUploadFileValidLocationName($dir, $sha1sum)
...
         }
 
         if (!is_dir(GLPI_DOC_DIR . "/" . $subdir)) {
-            Session::addMessageAfterRedirect(
+            Toolbox::logError(
                 sprintf(
                     __('Failed to create the directory %s. Verify that you have the correct permission'),
                     GLPI_DOC_DIR . "/" . $subdir
+                )
+            );
```
Comment (cedric-anne) [inline src/Document.php:1519]:
```suggestion
            trigger_error(
                sprintf(
                    __('Failed to create the directory %s. Verify that you have the correct permission'),
                    GLPI_DOC_DIR . "/" . $subdir
                ),
                E_USER_WARNING
            );
```

**Review by cedric-anne — CHANGES_REQUESTED** [review]

**Comment by cedric-anne** [comment]
> This PR introduces new strings to translate, I'm not sure it should be part of 10.0.6.

These messages are here from a very long time, so it could wait to be fixed in 10.0.7.

File: src/Document.php [hunk 9]
```diff
@@ -182,15 +182,22 @@ public function cleanDBonPurge()
...
-                        __('Succesful deletion of the file %s'),
-                        GLPI_DOC_DIR . "/" . $this->fields["filepath"]
+                        __('Successful deletion of the file %s'),
+                        $this->fields["filepath"]
                     ));
                 } else {
-                    Session::addMessageAfterRedirect(
+                    trigger_error(
                         sprintf(
                             __('Failed to delete the file %s'),
```
Comment (cedric-anne) [inline src/Document.php:191]:
```suggestion
                            'Failed to delete the file %s',
```

File: src/Document.php [hunk 10]
```diff
@@ -1157,8 +1164,12 @@ public function moveUploadedDocument(array &$input, $filename)
         }
 
         if (!is_file($fullpath)) {
-            Session::addMessageAfterRedirect(
+            trigger_error(
                 sprintf(__('File %s not found.'), $fullpath),
```
Comment (cedric-anne) [inline src/Document.php:1168]:
```suggestion
                sprintf('File %s not found.', $fullpath),
```

File: src/Document.php [hunk 11]
```diff
@@ -1186,17 +1197,24 @@ public function moveUploadedDocument(array &$input, $filename)
...
-                    __('Succesful deletion of the file %s'),
+                    __('Successful deletion of the file %s'),
                     $input['current_filename']
                 ));
             } else {
                // TRANS: %1$s is the curent filename, %2$s is its directory
-                Session::addMessageAfterRedirect(
+                trigger_error(
                     sprintf(
                         __('Failed to delete the file %1$s (%2$s)'),
```
Comment (cedric-anne) [inline src/Document.php:1207]:
```suggestion
                        'Failed to delete the file %1$s (%2$s)',
```

File: src/Document.php [hunk 12]
```diff
@@ -1257,8 +1275,12 @@ public static function moveDocument(array &$input, $filename)
         }
 
         if (!is_file($fullpath)) {
-            Session::addMessageAfterRedirect(
+            trigger_error(
                 sprintf(__('File %s not found.'), $fullpath),
```
Comment (cedric-anne) [inline src/Document.php:1279]:
```suggestion
                sprintf('File %s not found.', $fullpath),
```

File: src/Document.php [hunk 13]
```diff
@@ -1287,17 +1309,24 @@ public static function moveDocument(array &$input, $filename)
...
-                    __('Succesful deletion of the file %s'),
+                    __('Successful deletion of the file %s'),
                     $input['current_filename']
                 ));
             } else {
                // TRANS: %1$s is the curent filename, %2$s is its directory
-                Session::addMessageAfterRedirect(
+                trigger_error(
                     sprintf(
                         __('Failed to delete the file %1$s (%2$s)'),
```
Comment (cedric-anne) [inline src/Document.php:1319]:
```suggestion
                        'Failed to delete the file %1$s (%2$s)',
```

File: src/Document.php [hunk 14]
```diff
@@ -1388,17 +1417,24 @@ public static function uploadDocument(array &$input, $FILEDESC)
...
```
Comment (cedric-anne) [inline src/Document.php:1427]:
```suggestion
                        'Failed to delete the file %1$s (%2$s)',
```

File: src/Document.php [hunk 15]
```diff
@@ -1453,11 +1489,17 @@ public static function getUploadFileValidLocationName($dir, $sha1sum)
         }
 
         if (!is_dir(GLPI_DOC_DIR)) {
-            Session::addMessageAfterRedirect(
+            trigger_error(
                 sprintf(
                     __("The directory %s doesn't exist."),
```
Comment (cedric-anne) [inline src/Document.php:1494]:
```suggestion
                    "The directory %s doesn't exist.",
```

File: src/Document.php [hunk 16]
```diff
@@ -1471,16 +1513,23 @@ public static function getUploadFileValidLocationName($dir, $sha1sum)
...
-                GLPI_DOC_DIR . "/" . $subdir
+                $subdir
             ));
         }
 
         if (!is_dir(GLPI_DOC_DIR . "/" . $subdir)) {
-            Session::addMessageAfterRedirect(
+            trigger_error(
                 sprintf(
                     __('Failed to create the directory %s. Verify that you have the correct permission'),
```
Comment (cedric-anne) [inline src/Document.php:1523]:
```suggestion
                    'Failed to create the directory %s.',
```

**Review by cedric-anne — COMMENTED** [review]
We should not translate error messages in `trigger_error` calls, to be sure to have untranslated error log.

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**